tools/webpack/interactivity.js

```diff
@@ -0,0 +1,65 @@
+/**
+ * External dependencies
+ */
+const { join } = require( 'path' );
+
+/**
+ * Internal dependencies
+ */
+const { baseConfig } = require( './shared' );
+
+module.exports = {
+	...baseConfig,
+	watchOptions: {
+		aggregateTimeout: 200,
+	},
+	name: 'interactivity',
+	entry: {
+		index: {
+			import: `./packages/interactivity`,
```

Comment:
Here the config is seeking the entry point for `@wordpress/interactivity`, which is going to be read from `module` in `package.json` and resolved as `packages/interactivity/build-module/index.js`.

`build-module` folder contains files transpiled with Babel using the following command:

https://github.com/WordPress/gutenberg/blob/429d52edbb4e5da93e9636274616fd1f3fee16b3/package.json#L258

It uses the default Babel config: https://github.com/WordPress/gutenberg/blob/trunk/babel.config.js that gets some slight modifications depending on the target and you can check it in https://github.com/WordPress/gutenberg/blob/trunk/bin/packages/get-babel-config.js.

In effect, JSX gets replaced with 

https://github.com/WordPress/gutenberg/blob/429d52edbb4e5da93e9636274616fd1f3fee16b3/packages/babel-preset-default/index.js#L78-L84

That's why you see React pulled in.

Reply:
Thanks for taking a look, @gziolo! It works now without the `react` aliases. 😊